src/main/java/edu/cornell/kfs/module/cam/document/service/impl/CuAssetTransferServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+     * org.kuali.kfs.module.cam.document.AssetTransferDocument, org.kuali.kfs.coa.businessobject.Account,
+     * org.kuali.kfs.module.cam.businessobject.AssetPayment, boolean, org.kuali.kfs.module.cam.document.service.impl.AssetTransferServiceImpl.AmountCategory)
+     */
+    @Override
+    protected AssetGlpeSourceDetail createAssetGlpePostable(
+            AssetTransferDocument document, Account plantAccount, AssetPayment assetPayment, boolean isSource, AmountCategory amountCategory) {
+        AssetGlpeSourceDetail postable = super.createAssetGlpePostable(document, plantAccount, assetPayment, isSource, amountCategory);
+        clearSubAccountIfNecessary(postable);
+        return postable;
+    }
+
+    /**
+     * Clears out the sub-account number on the source detail if the account number matches a particular pattern.
+     * In the event that a badly-formatted pattern causes an exception while matching, the source detail will be left as-is.
+     * 
+     * This was separated into its own method for unit testing convenience. 
+     * 
+     * @param postable The GLPE source detail whose sub-account number may need to be cleared based on account number.
+     */
+    protected void clearSubAccountIfNecessary(AssetGlpeSourceDetail postable) {
+        try {
+            if (!shouldPreserveSubAccount(postable)) {
+                postable.setSubAccountNumber(null);
+            }
+        } catch (PatternSyntaxException e) {
+            LOG.warn("Invalid account number pattern found in parameter, this will cause the GLPE Source Detail's sub-account number to remain as-is", e);
+        }
+    }
+
+    /**
+     * Determines whether the sub-account number on a GLPE source detail should be preserved as-is,
+     * based upon the configuration of the "ASSET_TRANSFER_PLANT_ACCOUNTS_FOR_SUB_ACCOUNTS" parameter.
+     * The parameter contains a list of patterns that will be matched against the detail's account number,
+     * and its ALLOW/DISALLOW operator controls whether the patterns represent a whitelist or blacklist.
+     * 
+     * @param postable The GLPE source detail whose account number should be checked.
+     * @return true if the parameter is null or if its ALLOW/DISALLOW operator lines up with the match/no-match account check result, false otherwise.
+     * @throws PatternSyntaxException if the parameter exists and has a non-empty value but contains one or more invalid search expressions.
+     */
+    protected boolean shouldPreserveSubAccount(AssetGlpeSourceDetail postable) {
+        Parameter accountPatternsParameter = parameterService.getParameter(
+                Asset.class, CuCamsConstants.Parameters.ASSET_TRANSFER_PLANT_ACCOUNTS_FOR_SUB_ACCOUNTS);
+        if (ObjectUtils.isNull(accountPatternsParameter)) {
+            return true;
+        }
+        return EvaluationOperator.ALLOW.equals(accountPatternsParameter.getEvaluationOperator()) == accountMatchesAtLeastOnePattern(

```

Comment:
This line made me pause and spend a fair amount of time trying to understand what it was doing. It might be beneficial to pull this logic out into another method with a name that explains what's going on here just to make it more readable/understandable.